modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -438,8 +444,15 @@ cv::GStreamingCompiled cv::gimpl::GCompiler::compileStreaming()
 {
     // FIXME: self-note to DM: now keep these compile()/compileStreaming() in sync!
     std::unique_ptr<ade::Graph> pG = generateGraph();
-    GModel::Graph(*pG).metadata().set(Streaming{});
+    GModel::Graph gm(*pG);
+
+    gm.metadata().set(Streaming{});
     runPasses(*pG);
-    compileIslands(*pG);
+
+    std::shared_ptr<ade::Graph> gptr(gm.metadata().get<IslandModel>().model);
+    // Run topological sort on GIslandModel first
+    auto pass_ctx = ade::passes::PassContext{*gptr};
+    ade::passes::TopologicalSort{}(pass_ctx);
```

Comment:
Why do we need it here? Sorting the island model topologically should be the part of `runPasses()` - inst it?

Reply:
This code replaced to passes 
It moved to
`void topoSortIslands(ade::passes::PassContext &ctx);`
now